24_clustering.Rmd

```diff
@@ -17,9 +18,127 @@ features/samples that share pattern.  For example, with clustering, we
 can find group of samples that share similar community
 composition. There are multiple clustering algorithms available.
 
+## Introduction to bluster package
+
+*bluster* is a Bioconductor package specialized in clustering. It offers 
+multiple algorithms such as hierarchical clustering, DBSCAN, K-means, 
+amongst others. The first thing to do when using this package is to load 
+it. It's also possible to use `transformCounts` to transform the data 
+for the clustering.
+```{r load-bluster}
+# Load dependencies
+if (!require(bluster)) {
+  install.packages("bluster")
+  library(bluster)
+}
+
+# Optional: Apply transformation
+tse <- transformCounts(tse, method = "relabundance")
+```
+
+The clustering can be done on the features or the samples.
+
+### Samples clustering
+
+To cluster the samples, we simply need to transpose the assay so that 
+the samples are in rows.
+```{r sample-clust-prep}
+assay <- t(assay(tse, "relabundance"))
+```
+
+Then, to do the clustering, use the `clusterRows` function. Depending 
+on its parameters, it will perform a different algorithm. Here, we'll 
+perform a hierarchical clustering.
+```{r bluster-sample-hclust1}
+# Simple use of the hierarchical clustering which sets the cut height to
+# half the dendrogram height.
+hclust.out <- clusterRows(assay, HclustParam())
+
+# Add data to the TSE
+colData(tse)$clusters <- hclust.out
+
+# Checking the result
+hclust.out
+```
+
+Once the clustering on the samples is done, we can also plot the clusters.
+```{r bluster-sample-plot}
+library(scater)
+# Add the MDS dimensions for plotting
+tse <- runMDS(tse,
+  assay.type = "relabundance", FUN = vegan::vegdist,
+  method = "bray"
+)
+
+# Plot the clusters
+plotReducedDim(tse, "MDS", colour_by = "clusters")
+```
+
+To change the clustering parameters, simply change the parameters in 
+`HclustParam` function. To see the different parameters, check the 
+[HclustParam documentation](https://rdrr.io/github/LTLA/bluster/man/HclustParam-class.html).
+```{r bluster-sample-hclust2}
+# More complex use of the algorithm with another method and cut height
+hclust.out <- clusterRows(assay, HclustParam(method = "complete", cut.dynamic = TRUE))
+hclust.out
+```
```

Comment:
Check is failing because of this. This requires an extra package. I think this could be removed to avoid installation of an additional package